crates/server/src/admin/utils/auth.rs

```diff
@@ -61,14 +73,79 @@ pub fn verify_node_signature(
                 }
             };
 
+            let signature_str = match wallet_signature {
+                WalletSignature::String(sig) => sig,
+                _ => {
+                    return Err(ApiError {
+                        status_code: StatusCode::BAD_REQUEST,
+                        message: "Invalid wallet signature type.".into(),
+                    })
+                }
+            };
+
             if let Err(err) = verify_eth_signature(
                 &wallet_metadata.signing_key,
                 &payload.message.message,
-                wallet_signature,
+                signature_str,
             ) {
                 return Err(parse_api_error(err));
             }
 
+            Ok(true)
+        }
+        WalletType::SN { ref wallet_name } => {
+            let _sn_metadata: &StarknetSignatureMessageMetadata = match &payload.metadata {
+                SignatureMetadataEnum::SN(metadata) => metadata,
+                _ => {
+                    return Err(ApiError {
+                        status_code: StatusCode::BAD_REQUEST,
+                        message: "Invalid metadata.".into(),
+                    })
+                }
+            };
+
+            let (message_hash, signature) = match wallet_signature {
+                WalletSignature::StarknetPayload(payload) => {
+                    (payload.message_hash.clone(), payload.signature.clone())
+                }
+                _ => {
+                    return Err(ApiError {
+                        status_code: StatusCode::BAD_REQUEST,
+                        message: "Invalid wallet signature type for Starknet.".into(),
+                    })
+                }
+            };
+
+            let result = match wallet_name.as_str() {
+                "argentX" => tokio::task::block_in_place(|| {
+                    tokio::runtime::Runtime::new()
+                        .unwrap()
```

Comment:
Will this stop the node if it fails?

Reply:
No, this only blocks the current thread where it's executed. It does not interfere with other asynchronous tasks.

Reply:
I think it makes more sense if the function this is in is `async` instead ..._